### ISSUE TITLE
Submit all chat messages to results

### ISSUE THREAD

**Issue by DiGyt** [magpie-ea/magpie-base#21]
While we're at it already, you could help me with another thing ;)

When working with the Chat component, I want submit all the chat messages of one trial/chat to the results as a list (As shown below).

![chat_messages](https://user-images.githubusercontent.com/34550289/108065606-912fd580-705e-11eb-9041-fe4c2c243e2e.png)

However, my current method is super clumsy and I'm sure that there are way more elegant methods to solve this.
For my first tries, I tweaked the Chat component, such that I can reference the chatbox, and then extract all the messages from it:
```Javascript
methods: {
    submit_chat() {
      var container = [];
      for(var i=0; i<document.getElementById('chatbox_1').children.length; i++){
        var msg = document.getElementById('chatbox_1').children[i];
        var msg_text = msg.textContent;
        container.push(msg_text);
      }
      this.$magpie.addResult({ question: this.$magpie.currentTrial.QUD, answer: container});
      }
  }
```

Can you maybe give me a hint on how to access the `messages` variable from the Chat component directly/ how to improve this?
I think it would be nice to have an inbuilt function which does this automatically.

It would be also great to add the participant ID to this to distinguish between chat message authors, e.g. like:
```Javascript
container = [ "1 --- hey", "2 --- hello", "1 --- what's up", "2 --- nothing" ]
```

But I think this will be somewhat annoying with the method I'm currently using.,.

**Comment by marcelklehr** [comment]
Good point! I've just added an event to the component, so you can use it as follows:

```vue
	  <Screen>
        <template #0="{responses}">
          <Chat :messages.sync="responses.messages"></Chat>
          <button
            @click="
              $magpie.addTrialData(responses);
              $magpie.nextScreen();
            "
          >
            Next
          </button>
        </template>
      </Screen>
```

Note that this hasn't been released and that there have been significant changes to the API since the last release (eg. renaming addResult to addTrialData)

**Comment by DiGyt** [comment]
Great! Thanks @marcelklehr!

I'll look out for the next release then...

**Comment by DiGyt** [comment]
I mentioned some possible improvements for the given solution in #23 and #24, but the general problem of logging the chat component is solved with v0.3.0. Therefore I will close this Issue now...
